### ISSUE TITLE
Question on Figure 5 and on relaxation

### ISSUE THREAD

**Issue by EmreTaha** [Hhhhhhao/continuous_tokenizer#17]
Hi, 
i have 2 unrelated simple questions, it would be much appreciated if you can clarify them
* In figure 5, what are the other  points (non-pink ones)? 
* In terms of relaxation, have you experimented with [gumbel-softmax,](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html)? I am not aware of any application with Vq-Vae, but in reinforcement learning it has been used to avoid straight-through estimator when sampling from learnable discrete action embeddings.

Thanks in advance

**Comment by Hhhhhhao** [comment]
> Hi, i have 2 unrelated simple questions, it would be much appreciated if you can clarify them
> 
> * In figure 5, what are the other  points (non-pink ones)?
> * In terms of relaxation, have you experimented with [gumbel-softmax,](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html)? I am not aware of any application with Vq-Vae, but in reinforcement learning it has been used to avoid straight-through estimator when sampling from learnable discrete action embeddings.
> 
> Thanks in advance

Hi, for your questions
1. Pink indicates codewords, other gray points indicate data.
2. I have tried gumbel softmax before, it might have better generation performance but in terms of tokenizer, it is more difficult to tune/train.
